homeassistant/components/sabnzbd.py

```diff
@@ -0,0 +1,261 @@
+"""
+Support for monitoring an SABnzbd NZB client.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sabnzbd/
+"""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.discovery import SERVICE_SABNZBD
+from homeassistant.const import (
+    CONF_HOST, CONF_API_KEY, CONF_NAME, CONF_PORT, CONF_SENSORS,
+    CONF_SSL, CONF_SCAN_INTERVAL)
+from homeassistant.core import callback
+from homeassistant.helpers import discovery
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.event import async_track_time_interval
+from homeassistant.util.json import load_json, save_json
+
+REQUIREMENTS = ['pysabnzbd==1.0.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'sabnzbd'
+DATA_SABNZBD = 'sabznbd'
+
+_CONFIGURING = {}
+
+ATTR_SPEED = 'speed'
+BASE_URL_FORMAT = '{}://{}:{}/'
+CONFIG_FILE = 'sabnzbd.conf'
+DEFAULT_HOST = 'localhost'
+DEFAULT_NAME = 'SABnzbd'
+DEFAULT_PORT = 8080
+DEFAULT_SCAN_INTERVAL = timedelta(seconds=30)
+DEFAULT_SSL = False
+
+SERVICE_PAUSE = 'pause'
+SERVICE_RESUME = 'resume'
+SERVICE_SET_SPEED = 'set_speed'
+
+SIGNAL_SABNZBD_UPDATED = 'sabnzbd_updated'
+
+SENSOR_TYPES = {
+    'current_status': ['Status', None, 'status'],
+    'speed': ['Speed', 'MB/s', 'kbpersec'],
+    'queue_size': ['Queue', 'MB', 'mb'],
+    'queue_remaining': ['Left', 'MB', 'mbleft'],
+    'disk_size': ['Disk', 'GB', 'diskspacetotal1'],
+    'disk_free': ['Disk Free', 'GB', 'diskspace1'],
+    'queue_count': ['Queue Count', None, 'noofslots_total'],
+    'day_size': ['Daily Total', 'GB', 'day_size'],
+    'week_size': ['Weekly Total', 'GB', 'week_size'],
+    'month_size': ['Monthly Total', 'GB', 'month_size'],
+    'total_size': ['Total', 'GB', 'total_size'],
+}
+
+SPEED_LIMIT_SCHEMA = vol.Schema({
+    vol.Optional(ATTR_SPEED): cv.string,
+})
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_API_KEY): cv.string,
+        vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+        vol.Optional(CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL):
```

Comment:
@balloob do you think 30 seconds is a good default? I made it configurable because, for me, SABnzbd is idle probably 90% of the time and I would be happy only getting updates every 10 minutes or so. However, I can imagine some of the more "power users" would rather have more up-to-date download statistics for use in their automations to, for instance, determine when to reduce download speed or pause downloading altogether to prevent exceeding data caps. With the availability of gigabit+ residential connections, a 30 second update interval could mean downloading 3-4 GB between updates.